mailbox/maildir/src/main/java/org/apache/james/mailbox/maildir/mail/MaildirMailboxMapper.java

```diff
@@ -159,6 +160,42 @@ public boolean hasChildren(Mailbox mailbox, char delimiter) throws MailboxExcept
         return mailboxes.size() > 0;
     }
 
+    @Override
+    public MailboxId create(Mailbox mailbox) throws MailboxException {
+        Preconditions.checkArgument(mailbox.getMailboxId() == null, "A mailbox we want to create should not have a mailboxId set already");
+
+        MaildirId maildirId = MaildirId.random();
+        MaildirFolder folder = maildirStore.createMaildirFolder(mailbox);
+
+        if (!folder.exists()) {
+            boolean success = folder.getRootFile().exists();
+            if (!success) {
+                success = folder.getRootFile().mkdirs();
```

Comment:
please don't reuse success variable, it makes the code too complicated (from @mbaechler )